Python/auth-blocking-functions/main.py

```diff
@@ -0,0 +1,231 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from firebase_admin import auth, firestore, initialize_app
+from firebase_functions import identity_fn, https_fn
+
+import google.cloud.firestore
+
+initialize_app()
+
+
+# [START created_noop]
+@identity_fn.before_user_created()
+def created_noop(
+    event: identity_fn.AuthBlockingEvent,
+) -> identity_fn.BeforeCreateResponse | None:
+    return
+# [END created_noop]
+
+
+# [START signedin_noop]
+@identity_fn.before_user_signed_in()
+def signedin_noop(
+    event: identity_fn.AuthBlockingEvent,
+) -> identity_fn.BeforeSignInResponse | None:
+    return
+# [END signedin_noop]
+
+
+# [START v2ValidateNewUser]
+# [START v2beforeCreateFunctionTrigger]
+# Block account creation with any non-acme email address.
+@identity_fn.before_user_created()
+def validatenewuser(
+    event: identity_fn.AuthBlockingEvent,
+) -> identity_fn.BeforeCreateResponse | None:
+# [END v2beforeCreateFunctionTrigger]
+    # [START v2readUserData]
+    # User data passed in from the CloudEvent.
+    user = event.data
+    # [END v2readUserData]
+
+    # [START v2domainHttpsError]
+    # Only users of a specific domain can sign up.
+    if user.email is None or "@acme.com" not in user.email:
+        # Return None so that Firebase Auth rejects the account creation.
+        raise https_fn.HttpsError(
+            code=https_fn.FunctionsErrorCode.INVALID_ARGUMENT,
+            message="Unauthorized email",
+        )
+    # [END v2domainHttpsError]
+# [END v2ValidateNewUser]
+
+
+# [START setdefaultname]
+@identity_fn.before_user_created()
+def setdefaultname(
+    event: identity_fn.AuthBlockingEvent,
+) -> identity_fn.BeforeCreateResponse | None:
+    return identity_fn.BeforeCreateResponse(
+        # If no display name is provided, set it to "Guest".
+        display_name=(event.data.display_name if event.data.
+                      display_name is not None else "Guest"))
+# [END setdefaultname]
+
+
+# [START requireverified]
+@identity_fn.before_user_created()
+def requireverified(
+    event: identity_fn.AuthBlockingEvent,
+) -> identity_fn.BeforeCreateResponse | None:
+    if event.data.email is not None and not event.data.email_verified:
+        raise https_fn.HttpsError(
+            code=https_fn.FunctionsErrorCode.INVALID_ARGUMENT,
+            message="You must register using a trusted provider.",
+        )
+# [END requireverified]
+
+
+def send_verification_email_using_your_smtp_server(email, link):
+    return
+
+
+# TODO: Should really be non-blocking or client-side call.
```

Comment:
I don't think we should include this example.

1. There's no need for this to be blocking, AFAIK.
2. The (Python) Admin SDK intentionally doesn't support sending emails from Firebase, so there's a big hand-wave.

Reply:
I think we should remove or rework this example altogether, for both Python and Node, for the reasons given above.

Reply:
Roger that!  Saw your other PR (thumbs up emoji).